plugins/msg-broker/mcollective/lib/openshift/mcollective_application_container_proxy.rb

```diff
@@ -2029,9 +2029,26 @@ def move_gear(gear, destination_container, destination_district_uuid, change_dis
           end
         end
       else
-        # leave quota the same if node profile does not change
-        quota_blocks = Integer(gear_quota[3])
-        quota_files = Integer(gear_quota[6])
+        # if gear is over disk quota limit, increase destination quota by 1% so gear move, start, stop, restart complete
+        # get the base quota blocks and files destination container
+        destination_quota = destination_container.rpc_get_facts_direct(["quota_blocks", "quota_files"])
+        # set destination container quotas
+        quota_blocks = Integer(destination_quota[:quota_blocks])
+        quota_files = Integer(destination_quota[:quota_files])
+        # get current block and file usage from the source container
+        blocks_used = Integer(gear_quota[1])
+        if blocks_used >= quota_blocks
+          if blocks_used > quota_blocks * 1.01
+            log_debug "Cannot move #{gear.group_instance.gear_size} size gear to #{destination_node_profile}.  Gear exceeds quota limits and quota limit buffer also used up."
+            raise OpenShift::UserException.new("Error moving gear. Cannot move gear to #{destination_node_profile} node profile while gear exceeds profile quota limits.")
+          end
+          log_debug "WARNING: Blocks_used exceeds target disk quota, now using buffer"
+          quota_blocks = blocks_used * 1.01
```

Comment:
I'm pretty sure this algorithm can result in unbounded quota increase over repeated calls to `move_gear()` if `blocks_used` increments each time such that `blocks_used` is less than `quota_blocks * 1.01` (where `quota_blocks` is increased with each invocation). In fact, if the quota is applied with a hard limit, I wonder if `blocks_used > quota_blocks * 1.01` would always be false.

Consider the following three move attempts:

```
quota_blocks = 1000 (* 1.01 = 1010)
blocks_used = 1002
quota_blocks = 1012

quota_blocks = 1012 (* 1.01 = 1022)
blocks_used = 1011
quota_blocks = 1021

quota_blocks = 1021 (* 1.01 = 1031)
blocks_used = 1021
quota_blocks = 1031
```

I can imagine how this *might* be desirable since it gives each move operation a chance to progress, but if there's a process that's continuously writing to disk and some system process continuously trying to move the gear, there's now a hole for the process to write unbounded to storage.

Somebody care to verify my analysis?